java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -552,24 +552,20 @@ private boolean isArgumentsAcceptable(List<JavaType> argTypes, List<JavaType> fo
   }
 
   private boolean isAcceptableType(JavaType arg, JavaType formal, boolean autoboxing, boolean isParameterized) {
-    // FIXME SONARJAVA-1514 Inference of types for method using parameter types is not handled
-    if (!isParameterized && usesWildcard(formal)) {
+    // FIXME SONARJAVA-1298 type parameters should be handled in formal arguments prior to the call to this method
+    if (!isParameterized && (isParametrizedType(arg) || isParametrizedType(formal) || isWilcardType(arg) || isWilcardType(formal))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1067)